go/arrow/array/compare.go

```diff
@@ -735,10 +735,14 @@ func arrayApproxEqualFixedSizeList(left, right *FixedSizeList, opt equalOption)
 }
 
 func arrayApproxEqualStruct(left, right *Struct, opt equalOption) bool {
-	for i, lf := range left.fields {
-		rf := right.fields[i]
-		if !arrayApproxEqual(lf, rf, opt) {
-			return false
+	for i := 0; i < left.Len(); i++ {
+		if left.IsNull(i) {
+			continue
+		}
```

Comment:
shouldn't this be:

```go
if left.IsNull(i) {
    if !right.IsNull(i) {
        return false
    }
    continue
}
```
?

Reply:
we already checked for this in `baseArrayEqual`, see other `approx` implementations